code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -121,7 +123,13 @@
 
 /mob/living/silicon/robot/proc/is_component_functioning(module_name)
 	var/datum/robot_component/C = components[module_name]
-	return C && C.installed == 1 && C.toggled && C.is_powered()
+	return C && C.installed == 1 && C.toggled && C.is_powered() && !C.component_disabled
+
+/mob/living/silicon/robot/proc/disable_component(module_name, emp_duration)
```

Comment:
nitpicking, but you may just want to call the second argument "duration" in case we want to use this for something other than EMPs.